samanthas_telegram_bot/api_queries/send_get.py

```diff
@@ -0,0 +1,293 @@
+"""Functions for sending data to backend to create entities and get required data in return."""
+import json
+import logging
+
+import httpx
+from telegram import Bot, Update
+from telegram.constants import ParseMode
+
+from samanthas_telegram_bot.conversation.auxil.log_and_report import log_and_report
+from samanthas_telegram_bot.conversation.auxil.send_to_admin_group import send_to_admin_group
+from samanthas_telegram_bot.data_structures.constants import API_URL_PREFIX
+from samanthas_telegram_bot.data_structures.context_types import ChatData, UserData
+
+logger = logging.getLogger(__name__)
+
+STATUS_AT_CREATION_STUDENT_TEACHER = "awaiting_offer"
+STATUS_AT_CREATION_TEACHER_UNDER_18 = "active"
+
+
+# TODO how to best test these functions?  We can't test whether all data was collected by the bot,
```

Comment:
What you describe here sounds like E2E (or integration) testing, I'm not sure we need it right now, so better focus on unit testing. You can do some mocks on responses of external api and use them. 
Check out https://requests-mock.readthedocs.io/ library, probably something similar is available for httpx. But it's just to show example.

Reply:
Yep, I see there's something called [pytest_httpx](https://colin-b.github.io/pytest_httpx/). Ok, understood.